test/notifications/navigate_test.dart

```diff
@@ -0,0 +1,276 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/host/notifications.dart';
+import 'package:zulip/model/localizations.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+import 'package:zulip/notifications/navigate.dart';
+import 'package:zulip/notifications/receive.dart';
+import 'package:zulip/widgets/app.dart';
+import 'package:zulip/widgets/home.dart';
+import 'package:zulip/widgets/message_list.dart';
+import 'package:zulip/widgets/page.dart';
+
+import '../example_data.dart' as eg;
+import '../model/binding.dart';
+import '../test_navigation.dart';
+import '../widgets/dialog_checks.dart';
+import '../widgets/message_list_checks.dart';
+import '../widgets/page_checks.dart';
+
+Map<String, Object?> messageApnsPayload(
+  Message zulipMessage, {
+  String? streamName,
+  Account? account,
+}) {
+  account ??= eg.selfAccount;
+  return {
+    "aps": {
+      "alert": {
+        "title": "test",
+        "subtitle": "test",
+        "body": zulipMessage.content,
+      },
+      "sound": "default",
+      "badge": 0,
+    },
+    "zulip": {
+      "server": "zulip.example.cloud",
+      "realm_id": 4,
+      "realm_uri": account.realmUrl.toString(),
+      "realm_url": account.realmUrl.toString(),
+      "realm_name": "Test",
+      "user_id": account.userId,
+      "sender_id": zulipMessage.senderId,
+      "sender_email": zulipMessage.senderEmail,
+      "time": zulipMessage.timestamp,
+      "message_ids": [zulipMessage.id],
+      ...(switch (zulipMessage) {
+        StreamMessage(:var streamId, :var topic) => {
+          "recipient_type": "stream",
+          "stream_id": streamId,
+          if (streamName != null) "stream": streamName,
+          "topic": topic,
+        },
+        DmMessage(allRecipientIds: [_, _, _, ...]) => {
+          "recipient_type": "private",
+          "pm_users": zulipMessage.allRecipientIds.join(","),
+        },
+        DmMessage() => {"recipient_type": "private"},
+      }),
+    },
+  };
+}
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+  final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+
+  Future<void> init() async {
+    addTearDown(testBinding.reset);
+    testBinding.firebaseMessagingInitialToken = '012abc';
```

Comment:
Is this line needed? I'd think this data wouldn't be involved in any of these tests.